src/components/Categories/Categories.js

```diff
@@ -0,0 +1,150 @@
+import React, { Component } from "react";
+import task from '../../api/task';
+import auth from '../../api/auth';
+import categories from '../../api/category'
+
+export default class Categories extends Component {
+  state = {
+    categories: [],
+    allCategories: [],
+    currentCategory: 1,
+    showAddCategory: false,
+    error: null
+  }
+
+  /**
+  * Component setup
+  */
+  async componentDidMount() {
+    await this.getCategories()
+    this.loadAllCategories();
+  }
+
+  /**
+   * Get categories
+   * @return {Promise}
+   */
+  getCategories = async () => {
+    await this.setState({ categories: await task.getCategories(this.props.taskId) })
+  }
+
+  /**
+   * Load all categories
+   */
+  loadAllCategories = async () => {
+    if (auth.type() === "employer") {
+      this.setState({
+        allCategories: await categories.get()
+      })
+    }
+  }
+
+  /**
+   * Handle add categories event
+   */
+  handleAddCategoies = () => {
+    this.setState({ showAddCategory: !this.state.showAddCategory })
+  }
+
+  /**
+ * Sets state data when changes are made in text-inputs
+ * @param {Event} event
+ */
+  handleChange = event => {
+    this.setState({
+      [event.target.name]: event.target.value
+    })
+  }
+
+  /**
+  * Add a categories
+  */
+  submitAddCategori = async () => {
+    const categories = [...this.state.categories]
+    if (this.isIdInCategories(this.state.currentCategory)) {
+      return
+    }
+    await task.addCategory(this.props.taskId, this.state.currentCategory)
+    this.setState({ showAddCategory: false });
+    const title = this.state.allCategories[this.state.currentCategory - 1].title
+    categories.push({ title, id: categories.length + 1 })
```

Comment:
Use concat instead of push this makes an immutable copy of your categories.
```suggestion
    categories = categories.concat([{ title, id: categories.length + 1 }])
```